wordpress-mobile/gutenberg-mobile
# Use last gutenberg changes including the editor and block-editor packages refactoring

Requires https://github.com/WordPress/gutenberg/pull/14375

https://github.com/WordPress/gutenberg/pull/13088 introduced a breaking change for us, we need to update our dependency to the editor store and components as it (the editor package) has been split in 2 packages: `core/editor` and `core/block-editor` for more flexibility.

This PR updates our code to make it work with those changes:

- The main change of this PR is the switch from `core/editor` to `core/block-editor` selectors and actions in `BlockManager` along with the use of `BlockEditorProvider` in the parent component which makes sure both stores are in sync.

- In `AppContainer`, `MainApp`'s is removed and its logic is moved to `AppContainer` as the component we're using `BlockEditorProvider` handles defining the `SlotFillProvider` already.

- `editorSetup()` is simplified in `src/index.js` and most of its logic is moved to `edit-post/src/index.native.js`. To achieve this we needed to also move `UnsupportedBlock` to gutenberg.
 
- Styles (open for discussion): `src/colors.scss` and `src/variables.(ios|android).scss` are replaced with `src/_colors.scss` and `src/_variables.(ios|android).scss` and now supersede the files in `gutenberg/assets/stylesheets/`. This means that, just like with gutenberg, we don't need to explicitly have `@import "variables.scss";` or `@import "colors.scss";` in our stylesheets. This is closer to what we have on gutenberg web. However it also means that we need to redefine all necessary variables when we decide to have our own version of `src/_{file}.scss`.

- `@wordpress/block-editor`, `@wordpress/edit-post` and `@wordpress/token-list` are added to the list of symlinked packages

### Testing Instructions
- Try the demo app on both android and ios and notice that there is no regression
- Try running the app inside its parent app and notice there is no regression
